File: app/src/main/java/ar/com/develup/tateti/modelo/Partida.java
package ar.com.develup.tateti.modelo;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by maribelmai on 26/3/17.
 */

public class Partida {

    private transient String id;

    private String retador;
    private String oponente;
    private List<Movimiento> movimientos = new ArrayList<>();
    private String turno;
    private String ganador;

    public String getRetador() {
        return retador;
    }

    public void setRetador(String retador) {
        this.retador = retador;
    }

    public String getOponente() {
        return oponente;
    }

    public void setOponente(String oponente) {
        this.oponente = oponente;
    }

    public List<Movimiento> getMovimientos() {
        return movimientos;
    }

    public void setMovimientos(List<Movimiento> movimientos) {
        this.movimientos = movimientos;
    }

    public String getTurno() {
        return turno;
    }

    public void setTurno(String turno) {
        this.turno = turno;
    }

    public String getGanador() {
        return ganador;
    }

    public void setGanador(String ganador) {
        this.ganador = ganador;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Partida partida = (Partida) o;

        return id != null ? id.equals(partida.id) : partida.id == null;

    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }
}

File: app/src/main/java/ar/com/develup/tateti/modelo/Partida.kt
package ar.com.develup.tateti.modelo

import java.io.Serializable
import java.util.*

/**
 * Created by maribelmai on 26/3/17.
 */
data class Partida(@kotlin.jvm.JvmField var id: String? = null) : Serializable {
    var retador: String? = null
    var oponente: String? = null
    var movimientos: MutableList<Movimiento> = ArrayList()
    var turno: String? = null
    var ganador: String? = null

    fun calcularEstado() = when {
        oponente == null -> "ESPERANDO OPONENTE"
        ganador == null -> "EN JUEGO"
        else -> "PARTIDA FINALIZADA"
    }
}